File: C#/프로그래머스/2/42586. 기능개발/기능개발.cs
using System;
using System.Collections.Generic;

public class Solution {
    public int[] solution(int[] progresses, int[] speeds) {
        List<int> answer = new List<int>();
        Queue<int> queue = new Queue<int>();

        for (int i = 0; i < progresses.Length; i++) {
            int remain = 100 - progresses[i];
            int days = (int)Math.Ceiling((double)remain / speeds[i]);
            queue.Enqueue(days);
        }

        while (queue.Count > 0) {
            int cur = queue.Dequeue(); 
            int cnt = 1;

            while (queue.Count > 0 && queue.Peek() <= cur) {
                queue.Dequeue();
                cnt++;
            }

            answer.Add(cnt);
        }

        return answer.ToArray();
    }
}

File: 프로그래머스/2/42586. 기능개발/기능개발.cpp
#include <iostream>
#include <string>
#include <vector>
#include <queue>

using namespace std;

vector<int> solution(vector<int> progresses, vector<int> speeds) {
    vector<int> answer;
    queue<int> q;
    
    for(int p : progresses) {
        q.emplace(100 - p);
    }
    
    int idx = 0;
    while(!q.empty()) {
        int cnt = 0;
        
        while(!q.empty() && q.front() <= 0) {
            cnt++;   
            idx++;
            q.pop();
        }
        if(cnt > 0)
            answer.push_back(cnt);
        
        // 한 타임에 speeds_i만큼 줄이기
        if(!q.empty()) {
            for(int i = idx; i < (int)speeds.size(); ++i) {
                q.push(q.front() - speeds[i]);
                q.pop();
            }
        }
    }
    
    return answer;
}